great_expectations/experimental/datasources/file_path_data_asset.py

```diff
@@ -65,6 +79,16 @@ def __init__(self, **data):
         )
         self._all_group_names = self._regex_parser.get_all_group_names()
 
+    @property
+    def _base_directory(self) -> pathlib.Path:
```

Comment:
Why can't this just be a normal pydantic field?

Reply:
This one really nagged me. There are multiple issues going on here and now I'm seeing a `RuntimeWarning` due to the leading underscore field being passed to the dynamic assets.

1) A `DataAsset` can be instantiated without a `Datasource`, so we can't pull the `Datasource.base_directory` at instantiation time. Maybe this is wrong and we should be requiring a `datasource` in the `CSVAsset` args, because after all, what does it mean to have a `CSVAsset` without a `Datasource`? The property goes and looks for it when it is requested. One possible alternative solution would be to make this a `pydantic.PrivateAttr` and use a `Validator` for `datasource` that sets `_base_directory` upon `datasource` assignment. It is also important to note that `pydantic.Field`s [can't have leading underscores](https://github.com/pydantic/pydantic/issues/1476).
2) This is all complicated by the fact that changing this to `base_directory` from `_base_directory` and also updating the `FIELD_SUBSTITUTIONS` in `dynamic_pandas.py` to get rid the `RuntimeWarning` results in:
```
E   pydantic.error_wrappers.ValidationError: 1 validation error for _FilesystemDataAsset
E   type
E     field required (type=value_error.missing)
```
which is quite confusing. Maybe I'm just doing something wrong/missed a `base_directory` attribute somewhere, but I haven't figured it out.

Reply:
Agreed to handle this in Alex's PR, which leverages DataConnector here.